lib/utils/firebase_auth.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+
+class AuthProvider {
+  final FirebaseAuth _auth = FirebaseAuth.instance;
+
+  Future<bool> signInWithEmail(String email, String password) async{
+    try {
+      AuthResult result = await _auth.signInWithEmailAndPassword(email: email,password: password);
+      FirebaseUser user = result.user;
+      if(user != null)
+        return true;
+      else
+        return false;
+    } catch (e) {
+      print(e.message);
+      return false;
+    }
+  }
+
+  Future<void> logOut() async {
+    try {
+      await _auth.signOut();
+    } catch (e) {
+      print("error logging out");
+    }
+  }
+
+  Future<bool> loginWithGoogle() async {
+    try {
+      GoogleSignIn googleSignIn = GoogleSignIn();
+      GoogleSignInAccount account = await googleSignIn.signIn();
+      if(account == null )
+        return false;
+      AuthResult res = await _auth.signInWithCredential(GoogleAuthProvider.getCredential(
+        idToken: (await account.authentication).idToken,
```

Comment:
do this first, like: auto _auth = account.authentication;
then use _auth inside the brackets.
you calculate twice